crypto/signals.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "e_os.h"
+#include <string.h>
+
+#include <openssl/crypto.h>
+#include <internal/list.h>
+
+struct sigentry {
+    CRYPTO_SIGNAL p;
+    struct list list;
+};
+
+static struct list siglist;
+
+static ossl_inline int siglist_sigeq(struct list *l, void *data);
+
+static ossl_inline int siglist_sigeq(struct list *l, void *data)
+{
+    return container_of(l, struct sigentry, list)->p.signal == *(long*)data;
+}
+
+#if defined(OPENSSL_SYS_WINDOWS) || defined(_WIN32) || defined(__CYGWIN__)
+
+# include <windows.h>
+
+volatile PHANDLER_ROUTINE callback_handler = NULL;
+
+static BOOL WINAPI CRYPTO_SIGNAL_handler(DWORD dwType)
+{
+    if (list_empty(&siglist))
+        return FALSE;
+
+    struct list *l;
```

Comment:
since this is windows-only, maybe a variable declaration here is okay, but a project member will have to comment.

Reply:
Moved it... https://github.com/ckalina/openssl/commit/528e85bea64557627258abbb983a5317b7d4e4c8